[NXTFLASHER-888] add bootswitch feature to reset_netx command

gsathilscher opened the pull request:
Add a parameter to the reset command to make use of the NetX90 bootswitch feature.

Commit:
[code] fix wrong position of dirty-"+" in git version on dev branch

Commit:
[code] create first draft of bootswitch on normal reset command

dizzydevil commented on src/main.c:
```diff
@@ -870,12 +870,63 @@ static NETX_CONSOLEAPP_RESULT_T opMode_identify(void)
+
+			// Copy bootswitch image from INTRAM2 to INTRAM3.
+			// TODO this does not work. Find out why.
+			memcpy(bootswitch_image_src, bootswitch_image_dest, bootswitch_image_size);
```
the destination address comes first.

Commit:
[code] fix parmater order in memcpy in bootswitch

Commit:
[code] bootswitch: fix copy command
- send the length of the bootswitch binary to the target
- add a message on startup if a bootswitch was performed
- rename the parameter for bootswitch selection

Commit:
[code] refactor and add MFW bootswitch functionality
